### ISSUE TITLE
--last-failed and retest on success

### ISSUE THREAD

**Issue by carver** [joeyespo/pytest-watch#76]
I'd love some guidance on how to get `pytest --last-failed` to play nice with ptw. Right now, if the last failed tests succeeds, then ptw sits patiently until the next file change.

What I'd like is for it to run normally with last-failed, but when all of the last-failed tests succeed, then rerun all tests. Any suggestions?

**Comment by max-sixty** [comment]
Agree, this would be a great config. This would match the `--looponfail` behavior

**Comment by carver** [comment]
The workaround I've been using is `ptw mymodule/ tests/ -- --failed-first --exitfirst` in order to quickly re-loop on failing tests, but run all the remaining tests. ( or `ptw mymodule tests -- --ff -x` for the code golfers)

**Comment by max-sixty** [comment]
@carver that's a great idea.

To confirm - `-x` runs _one_ failed test, not all the failed tests; is that correct?

**Comment by carver** [comment]
`-x` halts at the first failed test, which may mean that multiple previously failing tests are run (if they start passing).

**Comment by max-sixty** [comment]
Yes, and then the next iteration runs only that single test, as opposed to looping over the failed tests?

**Comment by carver** [comment]
Say you have tests A, B, C, D, run in that order. Then you break tests B-D.

The next run would run:
- A - pass
- B - fail
- halt

Then you try to fix, but don't succeed. The next run is:
- B - fail
- halt

Then you fix B. The next run is:
- B - pass
- A - pass
- C - fail
- halt

**Comment by max-sixty** [comment]
Perfect, thanks for clarifying. 

For comparison, `--looponfail` would do:

- A - pass
- B - fail
- C - fail
- D - fail

Then you try to fix [B], but don't succeed. The next run is:

- B - fail
- C - fail
- D - fail

Then you fix B. The next run is:

- B - pass
- C - fail
- D - fail

...each run runs all the still-failing tests, until they're all fixed
